SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt

```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
+            ),
+            keyboardOptions = KeyboardOptions(
+                keyboardType = KeyboardType.Number,
+                imeAction = ImeAction.Done
+            ),
+            modifier = Modifier
+                .padding(top = 8.dp)
+                .padding(horizontal = 2.dp)
+                .height(56.dp)
+                .fillMaxWidth(),
+            shape = RoundedCornerShape(4.dp),
+            textStyle = TextStyle(
+                fontFamily = robotoFont,
+                fontWeight = FontWeight.Normal,
+                fontSize = 16.sp
+            ),
+            singleLine = true,
+            colors = TextFieldDefaults.textFieldColors(
+                textColor = Color.Black,
+                cursorColor = colorResource(id = R.color.primary_color),
+                backgroundColor = Color.Transparent,
+                focusedIndicatorColor = colorResource(id = R.color.list_prices_asset_component_code_color),
+                unfocusedIndicatorColor = colorResource(id = R.color.list_prices_asset_component_code_color),
+                disabledIndicatorColor = colorResource(id = R.color.list_prices_asset_component_code_color)
+            )
+        )
+    }
 
-                            AssetBankModel.Type.crypto -> {
+    @Composable
+    private fun CryptoCurrencyValueResult(
+        currencyState: MutableState<AssetBankModel>,
+        valueState: MutableState<String>,
+        valueAsset: MutableState<AssetBankModel>,
+        pairAsset: AssetBankModel,
+        typeOfValueState: MutableState<AssetBankModel.Type>
+    ) {
 
-                                codeAssetToUse = pairAsset
-                                val value = BigDecimal(stateInt).times(buyPriceDecimal)
-                                amount =  BigDecimalPipe.transform(value, pairAsset)!!
-                            }
+        val symbol = "${currencyState.value.code}-${pairAsset.code}"
+        val stateInt = valueState.value.toInt()
+        val buyPrice = listPricesViewModel?.getBuyPrice(symbol)
+        val buyPriceDecimal = BigDecimal(buyPrice?.buyPrice ?: 0)
+        var amount = "0"
+        var codeAssetToUse:AssetBankModel? = null
 
-                            AssetBankModel.Type.fiat -> {
+        when(typeOfValueState.value) {
 
-                                codeAssetToUse = currencyState.value
-                                val baseValue = AssetPipe.transform(
-                                    stateInt,
-                                    pairAsset,
-                                    "base"
-                                )
-                                val value = baseValue.divL(buyPriceDecimal)
-                                amount = value.toPlainString()
-                            }
+            AssetBankModel.Type.crypto -> {
 
-                            else -> {}
-                        }
+                valueAsset.value = currencyState.value
+                codeAssetToUse = pairAsset
```

Comment:
What's `valueAsset` and `codeAsset`?

Reply:
valueAsset its used to handle the label text in the ValueInput to get this kind of text: "Amount ETH" or "Amount BTC"

codeAsset its useed to set the currency code besie the operation string result

Reply:
Can we then use more descriptive variable names like `valueAssetLabel`?

Reply:
Sure, but its not a label, valueAsset its an AssetBankModel
then, i created the label and i created like this:


```
    val valueHint = buildAnnotatedString {
    append(stringResource(id = R.string.trade_flow_text_field_amount_placeholder))
    withStyle(style = SpanStyle(
        color = colorResource(id = R.color.list_prices_asset_component_code_color))) {
        append(" (${valueAsset.value.code})")
    }
}
```

If you see im using tha valueAsset as mutable (value member) and the the code of the model